flink-connectors/flink-connector-files/src/main/java/org/apache/flink/connector/file/sink/writer/FileWriter.java

```diff
@@ -85,17 +86,21 @@
 
     private final OutputFileConfig outputFileConfig;
 
+    private final Counter recordsOutCounter;
+
     /**
      * A constructor creating a new empty bucket manager.
      *
      * @param basePath The base path for our buckets.
+     * @param recordsOutCounter denoting the number records already written
      * @param bucketAssigner The {@link BucketAssigner} provided by the user.
      * @param bucketFactory The {@link FileWriterBucketFactory} to be used to create buckets.
      * @param bucketWriter The {@link BucketWriter} to be used when writing data.
      * @param rollingPolicy The {@link RollingPolicy} as specified by the user.
      */
     public FileWriter(
             final Path basePath,
+            final Counter recordsOutCounter,
```

Comment:
I propose to pass the whole metric group instead. We will need it anyways for advanced metrics.